### PR TITLE
Fixed memory leak in FloatComplexVectorList. Added functions to access data in rastermonochrome. 

### PR DESCRIPTION
**Pull request by kmartinpettersson**

### CODE REVIEW

File: source/draw/RasterMonochrome.ooc [hunk 1]
```diff
@@ -203,4 +204,27 @@ RasterMonochrome: class extends RasterPacked {
 		}
 		((this buffer pointer + y * this stride) as ColorMonochrome* + x)@ = value
 	}
+	getLine: func(y: Int) -> FloatVectorList {
+		result:= FloatVectorList new()
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: space before `:=` please

File: source/draw/RasterMonochrome.ooc [hunk 2]
```diff
@@ -203,4 +204,27 @@ RasterMonochrome: class extends RasterPacked {
...
+		version(safe) {
+			if (y > this size height || y < 0)
+				raise("Accessing RasterMonochrome index out of range in getLine")
+		}
+		for (x in 0..(this size width)) {
+				result add(this buffer pointer[y * this stride + x] as Float)
+		}
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: We don't need these brackets.

File: source/draw/RasterMonochrome.ooc [hunk 3]
```diff
@@ -203,4 +204,27 @@ RasterMonochrome: class extends RasterPacked {
...
+
+	getColon: func(x: Int) -> FloatVectorList {
+		result:= FloatVectorList new()
+		version(safe) {
+			if (x > this size width || x < 0)
+				raise("Accessing RasterMonochrome index out of range in getLine")
+		}
+		for (y in 0..(this size height)) {
+				result add(this buffer pointer[y * this stride + x] as Float)
+		}
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: Same here

File: source/draw/RasterMonochrome.ooc [hunk 4]
```diff
@@ -203,4 +204,27 @@ RasterMonochrome: class extends RasterPacked {
...
+				raise("Accessing RasterMonochrome index out of range in getLine")
+		}
+		for (x in 0..(this size width)) {
+				result add(this buffer pointer[y * this stride + x] as Float)
+		}
+		result
+	}
+
+	getColon: func(x: Int) -> FloatVectorList {
+		result:= FloatVectorList new()
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: Same here

File: source/draw/RasterMonochrome.ooc [hunk 5]
```diff
@@ -203,4 +204,27 @@ RasterMonochrome: class extends RasterPacked {
 		}
 		((this buffer pointer + y * this stride) as ColorMonochrome* + x)@ = value
 	}
+	getLine: func(y: Int) -> FloatVectorList {
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: space after `func`

File: source/draw/RasterMonochrome.ooc [hunk 6]
```diff
@@ -203,4 +204,27 @@ RasterMonochrome: class extends RasterPacked {
...
+			if (y > this size height || y < 0)
+				raise("Accessing RasterMonochrome index out of range in getLine")
+		}
+		for (x in 0..(this size width)) {
+				result add(this buffer pointer[y * this stride + x] as Float)
+		}
+		result
+	}
+
+	getColon: func(x: Int) -> FloatVectorList {
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: Same here

Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: "colon" is "tjocktarm", among other things. I think you mean "column". 

But, some consistency would be in order. If you have a `getColumn` function, the other one should be `getRow`, not `getLine`.

File: source/draw/RasterMonochrome.ooc [hunk 7]
```diff
@@ -203,4 +203,25 @@ RasterMonochrome: class extends RasterPacked {
...
+		version(safe) {
+			if (y > this size height || y < 0)
+				raise("Accessing RasterMonochrome index out of range in getLine")
+		}
+		for (x in 0..(this size width))
+				result add(this buffer pointer[y * this stride + x] as Float)
+		result
+	}
+
+	getColon: func (x: Int) -> FloatVectorList {
```
Comment (davidhesselbom) [inline source/draw/RasterMonochrome.ooc]: "colon" is "tjocktarm", among other things. I think you mean "column".

But, some consistency would be in order. If you have a getColumn function, the other one should be getRow, not getLine.

**Comment by kmartinpettersson** [comment]
Cleaned up, changed function names.

**Pull request merged**